### PR TITLE
feat(space-item): add active prop

### PR DESCRIPTION
**Pull request by andrewholsted**
This adds an activeSpaceId prop to the `react-component-spaces-list` and `react-component-space-item`. The idea being you pass it into the list itself and each space item checks if it's "active" or not. When it is, it adds an `.active` class to the component. 

Question:

I've never really used snapshots before. I wanted to explicitly assert that the `active` class was added to the component (since the first time my snapshot rendered, it wasn't). Is there a better way to do that?

Todo: Add tests for the spaces-list component? There's none presently.

### CODE REVIEW

File: packages/node_modules/@ciscospark/react-component-space-item/src/__snapshots__/index.test.js.snap [hunk 1]
```diff
@@ -1,5 +1,59 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
+exports[`SpaceItem component correctly renders an active space 1`] = `
+<div
+  className="space-item item active"
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/react-component-space-item/src/__snapshots__/index.test.js.snap]: Since the snapshot file has the `active` class, we don't need to verify it in the actual test.

File: packages/node_modules/@ciscospark/react-component-space-item/src/index.test.js [hunk 2]
```diff
@@ -44,6 +44,31 @@ describe('SpaceItem component', () => {
...
+        activeSpaceId="active-space-id"
+        onCallClick={onClick}
+        onClick={onClick}
+        name="Webex User"
+        teamColor="blue"
+        teamName="Best Team"
+      />
+    );
+    const component = renderer.getRenderOutput();
+    console.log(component.props.className)
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/react-component-space-item/src/index.test.js]: We'll need to remove this console.

File: packages/node_modules/@ciscospark/react-component-space-item/src/index.test.js [hunk 3]
```diff
@@ -44,6 +44,31 @@ describe('SpaceItem component', () => {
...
+    expect(component.props.className.includes('active')).toBeTruthy();
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/react-component-space-item/src/index.test.js]: This is the test we don't need due to the snapshot being valid.

File: packages/node_modules/@ciscospark/react-component-space-item/src/index.js [hunk 4]
```diff
@@ -91,7 +94,8 @@ function SpaceItem({
     <div
       className={classNames('space-item', styles.item, {
         [styles.hasCallSupport]: !!hasCallSupport,
-        [styles.isDecrypting]: !!isDecrypting
+        [styles.isDecrypting]: !!isDecrypting,
+        [styles.active]: activeSpaceId === id
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/react-component-space-item/src/index.js]: I think it might make more sense to change this prop to `isActive` and do the `activeSpaceId` comparison in the spaces list.

Comment (andrewholsted) [reply packages/node_modules/@ciscospark/react-component-space-item/src/index.js]: That works. Will update.

File: packages/node_modules/@ciscospark/react-component-space-item/src/index.test.js (hunk 3 above)
Comment (andrewholsted) [reply packages/node_modules/@ciscospark/react-component-space-item/src/index.test.js]: When I originally created the snapshot, it didn't have the `active` class. So my snapshot test was passing but it didn't really have everything I wanted. How do you know that the snapshot represents what you're expecting? Just manually validate and once you do that, you know it's good and you keep the snapshot?

Comment (adamweeks) [reply packages/node_modules/@ciscospark/react-component-space-item/src/index.test.js]: Gotcha, yea, when a snapshot is initially created, you have to manually verify it looks like you want it to.
